report/3.Theory/LanguageSpecification.tex

```diff
@@ -0,0 +1,92 @@
+numbers are represented using double-precision floating-point format, also known as the \textbf{\textit{double}} type.
+Booleans are represented using \textbf{true} or \textbf{false}.
+
+The collection types of the language are \textbf{\textit{toubles}} and \textbf{\textit{lists}} where:\\
+\textbf{\textit{toubles}} are collections of multiple types and \textbf{\textit{lists}} are collections of a single type.
+Dictionaries that functions as a collection of key-value pairs are deemed NTH and are not planned for implementation. 
+
+The language also includes the type \textbf{\textit{func}} which is a rename of the \textbf{\textit{lamda}} type.
+Also included is the type \textbf{\textit{none}} which is a rename of the \textbf{\textit{void}} type. 
+
+\textbf{Type Declaration}\\
+Types can be declared in the language in different ways.
+Firstly, types can be declared using the keyword \textbf{\textit{var}} (short for \textbf{variable\textit{}}) prefixed an assignment(line 1).
+Declarations can also be explicit by prefixing the type of the variable before an assignment(line 2), or by prefixing the type before the variable without an assignment(line 3).
+The three methods of declaration are seen in snippet below.
+\begin{lstlisting}
+var x = 5
+txt s = "string"
+bool b
+\end{lstlisting}
+
+Using the prefix \textbf{\textit{var}} enables the use of type-inference, where the type of the variable is automatically set during compile time \cite{typeinf}.
+Types in the language are type-static, meaning that once a type is set, it cannot be overwritten without an explicit valid type cast.
+
+\textbf{Method Declaration}\\
+The type of a methods parameters as well as its return type are explicitly declared when a method is declared. 
+The declaration is structured as an assignment where the methods name is declared on the left-hand-side prefixed by either \textbf{\textit{var}} or \textbf{\textit{func}}. 
+The parameters of the method is declared on the right-hand-side inside parenthesis followed by an arrow (\textbf{\textit{->}}) expressing the return type. 
+
+An example method that adds two numbers together and prints the value is shown below:
+\begin{lstlisting}
+var addNumbers = (num a, num b) -> none {
+	a + b print
+}
+\end{lstlisting} 
+A more explicit declaration of the same method can be seen in the snippet below:
+\begin{lstlisting}
+func<num,num,none> addNumbers = (num a, num b) -> none {
+	a + b print
+}
+\end{lstlisting}
+
+\textbf{Method Call}\\
+In order to better reflect the spoken English language, method calls are structured in a way where the first parameter to a method is written before the method-call itself.
+All subsequent parameters are written after the method-call and are delimited by the use of white-spacing and can be optionally grouped inside parenthesis.
+When methods are passed in as parameters, they shall always be prefixed with '::' characters to differentiate between methods as parameters and other method calls. 
```

Comment:
We talked about not having the "::" operator. However, now that I read our initial intention of a clear distinction between method invoking and methods as parameters I am reconsidering. I think we should talk about it in the group.